src/disjoint_set.jl

```diff
@@ -107,17 +107,21 @@ end
 
 @compat type DisjointSets{T}
     intmap::Dict{T,Int}
+    revmap::Dict{Int,T}
```

Comment:
Agreed, this performance-wise better. I've updated my pull to accommodate this change.